v0/destinations/one_signal/transform.js

```diff
@@ -0,0 +1,275 @@
+const get = require("get-value");
+const { EventType } = require("../../../constants");
+const {
+  ConfigCategory,
+  mappingConfig,
+  BASE_URL,
+  ENDPOINTS
+} = require("./config");
+const {
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  constructPayload,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  getIntegrationsObj,
+  CustomError,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  getDestinationExternalID
+} = require("../../util");
+
+const responseBuilder = (payload, endpoint) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = `${BASE_URL}${endpoint}`;
+    response.headers = {
+      Accept: "application/json",
+      "Content-Type": "application/json"
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+};
+
+const populateDevicetype = (message, payload) => {
+  const integrationsObj = getIntegrationsObj(message, "one_signal");
+  const devicePayload = payload;
+  if (integrationsObj && integrationsObj.deviceType) {
+    devicePayload.device_type = integrationsObj.deviceType;
+    devicePayload.identifier = integrationsObj.identifier;
+  }
+};
+const getExternalIdentifiersMapping = message => {
+  // Data Structure expected:
+  // context.externalId: [ {type: playerId, id: __id}]
+  const externalId = get(message, "context.externalId");
+  let playerId;
+  if (externalId && Array.isArray(externalId)) {
+    externalId.forEach(id => {
+      if (id.type === "playerId") {
```

Comment:
```suggestion
      if (id.type === "onesignalPlayerId") {
```
We generally add destination name with the type.

Reply:
removed